drop-transfer/src/ws/server/mod.rs

```diff
@@ -861,7 +866,11 @@ impl FileXferTask {
             let size = tmp_size.unwrap_or(0);
             events.checksum_start(size).await;
 
-            Some(|progress| events.checksum_progress(progress))
+            Some(|progress| async move {
+                if (progress / CHECKSUM_CHUNK_SIZE as u64) % checksum_events_granularity == 0 {
```

Comment:
I think it might not work as expected because you cannot assume that `progress` is a multiple of `CHECKSUM_CHUNK_SIZE`. This means the logic should be more focused on accumulating progress
```rust
from_last_event += progress;
if from_last_event > granularity {
    fire_event();
    from_last_event -= granularity;
}
```

Reply:
Now I made a mistake, progress is already accumulated in `progress` so it should be something like this:
```rust
if progress - last > granularity {
    last = progress - (progress % granularity);
    fire_event();
}
```